automatic/nodejs.install/update.ps1

```diff
@@ -33,35 +33,40 @@ function global:au_SearchReplace {
 }
 
 function global:au_GetLatest {
-  $urlsToGrabMsis = @(
-    "https://nodejs.org/en/download"
-    "https://nodejs.org/en/download/current"
-  )
-
-  $lts_page = Invoke-WebRequest -Uri "https://github.com/nodejs/Release/blob/master/README.md" -UseBasicParsing
+  $scheduleUri = 'https://raw.githubusercontent.com/nodejs/Release/main/schedule.json'
+  $schedules = Invoke-RestMethod -Uri $scheduleUri -UseBasicParsing
+
+  $curDate = (Get-Date).Date
+  $supportedChannels = @()
+  $schedules.PSObject.Properties.Name | ForEach-Object {
+    $name = $_
+    $schedule = $schedules.$name
+    $scheduleStart = [datetime]$schedule.start
+    $scheduleEnd = [datetime]$schedule.end
+    if (($scheduleStart -le $curDate) -and ($scheduleEnd -ge $curDate)) {
+      $supportedChannels += $name
+    }
+  }
 
-  $urlsToGrabMsis += $lts_page.links | Where-Object href -match "\/latest\-v.*\/$" | Select-Object -expand href
+  $versionsUri = 'https://nodejs.org/dist/index.json'
+  $versions = Invoke-RestMethod -Uri $versionsUri -UseBasicParsing
 
   $streams = @{ }
 
-  $urlsToGrabMsis | ForEach-Object {
-    $uri = $_
-    $download_page = Invoke-WebRequest -Uri $uri -UseBasicParsing
-
-    $msis = $download_page.links | Where-Object href -match '\.msi$' | Select-Object -expand href | ForEach-Object {
-      if (!$_.StartsWith('http')) { return $uri + $_ } else { $_ }
-    }
-
-    $url32 = $msis | Where-Object { $_ -match 'x86' } | Select-Object -first 1
-    $version = $url32 -split '\-v?' | Select-Object -last 1 -skip 1
-    $versionMajor = $version -replace '(^\d+)\..*', "`$1"
-    if ($streams.ContainsKey($versionMajor)) { return ; }
-
-    $url64 = $msis | Where-Object { $_ -match "\-x64" } | Select-Object -first 1
+  $supportedChannels | ForEach-Object {
+    $channel = $_
+    $latestVersion = $versions | Where-Object -FilterScript { $_.version.StartsWith($channel) } | Select-Object -First 1
+    $version = $latestVersion.version
+    $versionStrict = [version]::Parse($latestVersion.version.Substring(1))
 
-    if ($url32 -eq $url64) { throw "The 64bit executable is the same as the 32bit" }
+    $url32 = "https://nodejs.org/dist/$version/node-$version-x86.msi"
+    $url64 = "https://nodejs.org/dist/$version/node-$version-x64.msi"
 
-    $streams.Add($versionMajor, @{ Version = $version ; URL32 = $url32; URL64 = $url64 } )
+    $streams.Add($versionStrict.Major.ToString(), @{
+        Version = $versionStrict.ToString()
+        URL32   = $url32
+        URL64   = $url64
+      })
```

Comment:
Adding it like this can have the issue of throwing an exception if the key already exist (for instance, two items with the same major version is listed on the website).

The old code already had a check on whether the key already is present or not because of this happening in the past.

Please add some error handling for this, either by just skipping the adding of the stream if the key is already present, updating the stream, or something else.